client/src/app/(modules)/network/(form)/new/organisation/page.tsx

```diff
@@ -0,0 +1,328 @@
+        label: theme?.name,
+        value: theme?.id?.toString(),
+      })),
+    },
+    short_description: {
+      label: 'Brief description',
+      required: true,
+      zod: z.string().nonempty('Field is required').max(350, {
+        message: 'Brief description is limited to 350 characters.',
+      }),
+      type: 'textarea',
+      maxSize: 350,
+    },
+    description: {
+      label: 'Extended description',
+      zod: z
+        .string()
+        .max(3000, {
+          message: 'Extended description is limited to 3000 characters.',
+        })
+        .optional(),
+      type: 'textarea',
+      maxSize: 3000,
+    },
+    country: {
+      label: 'Country',
+      zod: z
+        .enum(countries?.map((type) => type?.id?.toString()) as [string, ...string[]])
+        .optional(),
+      type: 'select',
+      options: countries.map((country) => ({
+        label: country?.name,
+        value: country?.id?.toString(),
+      })),
+    },
+    url: {
+      label: 'Website',
+      zod: z
+        .string()
+        .max(255, {
+          message: 'Organisation name is limited to 255 characters.',
+        })
+        .regex(new RegExp('^(https?:\\/\\/)?(www\\.)?[a-zA-Z0-9-]+(\\.[a-zA-Z0-9-]+)+$'), {
+          message: 'Please, enter a valid URL.',
+        })
+        .max(255, {
+          message: 'Website is limited to 255 characters.',
+        })
+        .optional(),
+      type: 'text',
+      maxSize: 255,
+      description: (
+        <div className="leading-normal text-gray-500">
+          Accepted URLs format:
+          <ul className="ml-4 list-disc">
+            <li>https://irc-orcasa.eu/ or https://www.irc-orcasa.eu/</li>
+            <li>www.irc-orcasa.eu/</li>
+            <li>irc-orcasa.eu/</li>
+          </ul>
+        </div>
+      ),
+    },
+  };
+  const formSchema = z
+    .object(
+      (fields ? Object.keys(fields) : []).reduce((acc: { [key: string]: Field['zod'] }, field) => {
+        if (fields) {
+          acc[field] = fields[field].zod;
+        }
+        return acc;
+      }, {}),
+    )
+    .superRefine(({ organization_type, organization_type_other }, refinementContext) => {
+      if (organization_type === OtherId && typeof organization_type_other === 'undefined') {
+        return refinementContext.addIssue({
+          code: 'custom',
+          message: 'Organisation type: Other is mandatory when the organisation type is Other.',
+          path: ['organization_type_other'],
+        });
+      }
+      return refinementContext;
+    });
+
+  const form = useForm<z.infer<typeof formSchema>>({
+    resolver: zodResolver(formSchema),
+  });
+
+  const router = useRouter();
+
+  if (!hasData || !fields) {
+    return null;
+  }
+
+  const onSubmit: SubmitHandler<z.infer<typeof formSchema>> = (data) => {
+    const normalizedData = {
+      // ...data,
+      url: data.url && data.url.startsWith('http') ? data.url : `https://${data.url}`,
```

Comment:
If `data.url` is `undefined`, then you're sending `'https://undefined'` to the server.